src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/repositories/ReportRunRepository.kt

```diff
@@ -0,0 +1,48 @@
+package org.vaccineimpact.orderlyweb.db.repositories
+
+import com.google.gson.Gson
+import org.vaccineimpact.orderlyweb.db.*
+import java.sql.Timestamp
+import java.time.Instant
+
+interface ReportRunRepository
+{
+    fun addReportRun(
+        key: String,
+        user: String,
+        date: Instant,
+        report: String,
+        instances: Map<String, String>,
+        params: Map<String, String>,
+        gitBranch: String?,
+        gitCommit: String?
+    )
+}
+
+class OrderlyWebReportRunRepository : ReportRunRepository
+{
+    override fun addReportRun(
+        key: String,
+        user: String,
+        date: Instant,
+        report: String,
+        instances: Map<String, String>,
+        params: Map<String, String>,
+        gitBranch: String?,
+        gitCommit: String?
+    )
+    {
+        JooqContext().use {
+            it.dsl.insertInto(Tables.ORDERLYWEB_REPORT_RUN)
```

Comment:
What happens if a user updates the parameters for run report and reruns ? shouldn't we have a where clause if we want changes to be accommodated. Is there a different plan for this scenario?

We can check the DB if an instance of the report has been run and update the db tables where necessary.

Reply:
Every re-run gets a new key and a new report version so we can trust that for every fresh run, we're inserting a new row into this table, not updating an existing one.

Reply:
We currently consider changed parameters means a new run. This is consistent with orderly.server, which will assign a new the job a new `key`. Hope that makes sense? But it's a good question - I think that running the same report with different parameters (i.e. batching) is part of the workflows discussion - see VIMC-4570.